lib/topologies/replset_state.js

```diff
@@ -263,6 +265,21 @@ ReplSetState.prototype.update = function(server) {
     return false;
   }
 
+  // Update logicalSessionTimeoutMinutes
+  if (ismaster.logicalSessionTimeoutMinutes !== undefined) {
+    if (
+      self.logicalSessionTimeoutMinutes === undefined ||
+      ismaster.logicalSessionTimeoutMinutes === null
+    ) {
+      self.logicalSessionTimeoutMinutes = ismaster.logicalSessionTimeoutMinutes;
+    } else {
+      self.logicalSessionTimeoutMinutes = Math.min(
```

Comment:
so it says in specs "it MUST set TopologyDescription.logicalSessionTimeoutMinutes to the smallest logicalSessionTimeoutMinutes value among all ServerDescriptions of known ServerType". 

if this is just taking the minimum of the master and itself, will that accomplish that? (maybe it is communicated to other nodes through the master?)

Reply:
Yeah there was definitely two ways to go about this: 
  - track each incoming `logicalSessionTimeoutMinutes` from ismaster responses for connected replset members, and then on request of the value iterate through all and find the lowest one

  - track each `logicalSessionTimeoutMinutes` as they come in, and only ever store it once

Unless you're seeing something wrong with the logic, the intent here is to "pre-compute" the smallest known value

Reply:
ahhh ok I see! I was thinking that only the master itself sends ismaster responses. but now that it look into it I see that all the members send them. makes more sense